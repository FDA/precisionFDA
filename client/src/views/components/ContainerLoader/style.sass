.pfda-container-loader
  position: relative

  &__container
    text-align: center
    position: absolute
    top: 0
    left: 0
    width: 100%
    margin-top: 10%

  &__text
    margin-bottom: 30px
    font-size: 20px
