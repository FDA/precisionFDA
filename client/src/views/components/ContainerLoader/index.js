import React from 'react'
import PropTypes from 'prop-types'

import Loader from '../Loader'
import './style.sass'


const ContainerLoader = ({ text }) =>
  <div className="pfda-container-loader">
    <div className="pfda-container-loader__container">
      {(text) && <div className="pfda-container-loader__text">{text}</div>}
      <div className="pfda-container-loader__loader">
        <Loader />
      </div>
    </div>
  </div>

ContainerLoader.propTypes = {
  text: PropTypes.string,
}

export default ContainerLoader
