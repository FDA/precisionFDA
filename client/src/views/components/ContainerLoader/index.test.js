import React from 'react'
import { shallow } from 'enzyme'

import ContainerLoader from '.'


describe('ContainerLoader', () => {
  it('matches snapshot', () => {
    const component = shallow(<ContainerLoader />)
    expect(component).toMatchSnapshot()
  })
})

describe('ContainerLoader has text', () => {
  it('renders loader without text', () => {
    const wrapper = shallow(<ContainerLoader />)
    expect(wrapper.exists('.pfda-container-loader__text')).toEqual(false)
  })

  it('renders loader with text', () => {
    const wrapper = shallow(<ContainerLoader text="Some Text" />)
    expect(wrapper.exists('.pfda-container-loader__text')).toEqual(true)
    expect(wrapper.find('.pfda-container-loader__text').at(0).text()).toEqual('Some Text')
  })
})
