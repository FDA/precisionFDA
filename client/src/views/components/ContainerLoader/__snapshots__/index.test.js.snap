// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerLoader matches snapshot 1`] = `
<div
  className="pfda-container-loader"
>
  <div
    className="pfda-container-loader__container"
  >
    <div
      className="pfda-container-loader__loader"
    >
      <Loader />
    </div>
  </div>
</div>
`;
